geps/gep-3155/index.md

```diff
@@ -0,0 +1,205 @@
+backend. This adds that configuration to both Gateway and Service (via 
+BackendTLSPolicy).
+
+Specifying credentials at the service level allows to provide different
+credentials to connect to specific backend. This allows to address a use case,
+when destination service owner requires their clients to use a dedicated client
+certificate.
+
+**1. Add a new `BackendTLS` field at the top level of Gateways**
+
+```go
+type GatewaySpec struct {
+  // BackendTLS configures TLS settings for when this Gateway is connecting to
+  // backends with TLS.
+  BackendTLS GatewayBackendTLS `json:"backendTLS,omitempty"'
+}
+
+type GatewayBackendTLS struct {
+  // ClientCertificateRef is a reference to an object that contains a Client
+  // Certificate.
+  //
+  // References to a resource in different namespace are invalid UNLESS there
+  // is a ReferenceGrant in the target namespace that allows the certificate
+  // to be attached. If a ReferenceGrant does not allow this reference, the
+  // "ResolvedRefs" condition MUST be set to False for this listener with the
+  // "RefNotPermitted" reason.
+  //
+  // ClientCertificateRef can reference to standard Kubernetes resources, i.e.
+  // Secret, or implementation-specific custom resources.
+  //
+  // This setting can be overriden on the service level by use of BackendTLSPolicy.
+  ClientCertificateRef SecretObjectReference `json:"clientCertificateRef,omitempty"`
+}
+```
+
+**2. Add a new `Client` field at the top level of BackendTLSPolicy**
+
+```go
+type BackendTLSPolicySpec {
+  // Client specifies overriden TLS settings for the given service.
+  //
+  // If Client is omitted, backend mTLS settings inherited from the given gateway are
+  // used.
+  Client TLSClientSettings `json:"client"`
+}
+
+type TLSClientSettings struct {
+  // CertificateRef is a reference to an object that contains a Client
+  // Certificate.
+  //
+  // References to a resource in different namespace are invalid UNLESS there
+  // is a ReferenceGrant in the target namespace that allows the certificate
+  // to be attached. If a ReferenceGrant does not allow this reference, the
+  // "ResolvedRefs" condition MUST be set to False for this listener with the
+  // "RefNotPermitted" reason.
+  //
+  // CertificateRef can reference to standard Kubernetes resources, i.e.
+  // Secret, or implementation-specific custom resources.
+  //
+  // If CertificateRef is not specified, no client certificate is presented when
+  // connecting to the service.
+  CertificateRef SecretObjectReference `json:"certificateRef,omitempty"`
+}
+```
+
+#### Limitations
+
+Configuring client certificate on the service level may result in multiple Gateways
+from different vendors sharing the same identity when connecting to a single service.
+
+The problem could be partially mitigated by allowing shared policies to selectively 
+target specific Gateway instances, GatewayClasses or gateway namespaces.
```

Comment:
This limitation is why we did not include client certificate configuration in the initial version of BackendTLSPolicy. Similarly to @mikemorris' comment above, I'd like some use cases that illustrate the need for this, as I think that it's far more likely to be used incorrectly than only having the TLS config for a Gateway present on that Gateway.

To be honest, the BackendTLSPolicy is supposed to be owned by the same party as the Service, and I question if having the party owning the server side of the transaction also specify the client keypair to be used is really fulfilling the spirit of what mutual TLS is supposed to be about.

To put this another way - if you lock your front door, but leave the key tied with a string to the door, are you increasing or decreasing security? Part of the point of adding the second factor of client authentication to a mutual TLS transaction is to ensure that there's some checks and balances in who gets a valid client certificate, and this feels like it bypasses that.

Reply:
Added the use cases, added the write-up on security of this solution, can you PTAL at the updated doc?